guardian/dotcom-rendering
# Apps Rendering: Refactor out `React.FC<Props>` pattern

## What does this change?

- Refactors out the `React.FC<Props>` pattern because it doesn't provide any clear benefits and it's stopping us from making a full migration to Storybook 8 (see #10917)
- Adds an `eslint` rule against using `FC`-style components (thx @mxdvl for the idea)

## How to test?

Everything should look the same, so:

- Check storybook has no meaningful diffs
- Deploy to CODE & check

Closes https://github.com/guardian/dotcom-rendering/issues/11633
